### DISCUSSION TITLE
Why Panopticon changing secret word in Akeeba Backup?

### DISCUSSION THREAD

**Discussion by altairtg** [akeeba/panopticon#579] [General]
Hello, when I get an error that Akeeba Backup is not connected, I click the relink. After this operation, the Akeeba Backup Secret Word is changed to a new one in the Akeeba Backup panel, why? I have CRON jobs with an old entry and I don't want to change it. If I enter the old value again, Panopticon still informs that there is an error.

**Comment by nikosdion** [comment]
It will change the secret word in exactly two circumstances:
1. None was set up to begin with.
2. The one set up was insecure, therefore Akeeba Backup did not allow you to use it to access its remote JSON API anymore.

**Reply by altairtg** [reply]
"1. None was set up to begin with."
Secret word was set long time ago to CRON Jobs
"2. The one set up was insecure, therefore Akeeba Backup did not allow you to use it to access its remote JSON API anymore"
How to understand? Previously it was automatically generated by Akeeba Backup and it was OK. Why can't I type the previous Secret Word? I don't want to manually change CRON tasks in 70 installations.

**Reply by nikosdion** [reply]
> Why can't I type the previous Secret Word?

Because that would be stupid, on account of having the _Panopticon connector_ on your site which can collect this information and communicate it back to Panopticon. And I think that's where you messed up.

When you first set up a site in Panopticon you must ONLY enter its name, endpoint URL, and token. Nothing more, nothing less. Panopticon then tries to connect to the site.

When it succeeds, you can either wait up to 15' for the site's information to be updated **OR** you can click on the update button in the Joomla version panel of the site overview page. Part of the information conveyed is whether Akeeba Backup is installed, whether it's the Pro version, whether the JSON API is enabled, and what is the JSON API key.

If you are too impatient this information is not yet retrieved from the site and you click the reconnect button. In this case Panopticon thinks that your site does not have an Akeeba Backup JSON API key and creates a new one. Therefore, you shoot your feet.

If this information has already been retrieved, there are two cases when Panopticon needs to regenerate a key. I already told you what they are.

So, assuming that NEITHER of the two cases I explained is true, the answer to your complaint is that you are too impatient. So, maybe slow down and try to treat other people a bit kinder? Just maybe?

**Reply by altairtg** [reply]
Thanks nicosdion for the detailed explanation :)
To sum up, if I see that there is no connection to Akeeba Backup PRO, should I not click reconnect and wait 15 minutes?

Regarding my impatience, I am rather calm and patient :)
And by the way, you write so that I can treat people better, I didn't write anything like that...
btw I'm using Akeeba PRO from 2017 :)

**Reply by nikosdion** [reply]
If there is no connection and you _think_ there should be a connection, click on the update button in the site overview. That one here:

<img width="417" alt="image" src="https://github.com/akeeba/panopticon/assets/256041/3389c12a-0ae3-4552-99dd-24848cfb13c9">

This will reload the site information, just like what happens automatically every 15 minutes. If you still get no connection with Akeeba Backup, click on the reconnect button and see what happens. Panopticon tries to determine the problem and solve it.

Remember, there is no software which can predict, identify, and solve 100% of possible problems. It may be that on a very small number of sites there is no problem we have anticipated, we brute force a new connection, and it works. If that's the case, you get to live with it. That's realistically the best we can do. The alternative would be for Panopticon to keel over and play dead.
